integration-tests/http/__tests__/product/store/product.spec.ts

```diff
@@ -1928,6 +1928,130 @@ medusaIntegrationTestRunner({
         )
       })
 
+      it("should retrieve product withhout category if the categories field is passed", async () => {
+        const [product] = await createProducts({
+          title: "test category prod",
+          status: ProductStatus.PUBLISHED,
+          options: [{ title: "size", values: ["large"] }],
+          variants: [
+            {
+              title: "test category variant",
+              options: { size: "large" },
+              prices: [
+                {
+                  amount: 3000,
+                  currency_code: "usd",
+                },
+              ],
+            },
+          ],
+        })
+
+        await api.post(
+          `/admin/sales-channels/${defaultSalesChannel.id}/products`,
+          { add: [product.id] },
+          adminHeaders
+        )
+
+        const response = await api.get(
+          `/store/products/${product.id}?fields=*categories`,
+          storeHeaders
+        )
+
+        expect(response.status).toEqual(200)
+        expect(response.data.product).toEqual(
+          expect.objectContaining({
+            id: product.id,
+            categories: [],
+          })
+        )
+      })
+
+      it("should retrieve product with category", async () => {
+        const [product] = await createProducts({
+          title: "test category prod",
+          status: ProductStatus.PUBLISHED,
+          options: [{ title: "size", values: ["large"] }],
+          variants: [
+            {
+              title: "test category variant",
+              options: { size: "large" },
+              prices: [
+                {
+                  amount: 3000,
+                  currency_code: "usd",
+                },
+              ],
+            },
+          ],
+        })
+
+        const category = await createCategory(
+          { name: "test", is_internal: false, is_active: true },
+          [product.id]
+        )
+
+        await api.post(
+          `/admin/sales-channels/${defaultSalesChannel.id}/products`,
+          { add: [product.id] },
+          adminHeaders
+        )
+
+        const response = await api.get(
+          `/store/products/${product.id}?fields=*categories`,
+          storeHeaders
+        )
+
+        expect(response.status).toEqual(200)
+        expect(response.data.product).toEqual(
+          expect.objectContaining({
+            id: product.id,
+            categories: [expect.objectContaining({ id: category.id })],
+          })
+        )
+      })
+
+      it("should return 404 for a product in a disabled category", async () => {
```

Comment:
discussion: should a product be returned in this case as well? what about case where a product belongs to multiple categories where some of them are active

Reply:
I agree with you this is why I added the comment. Added the test since this is the current behaviour and I wasn't sure if it was intentional but on second tought looks like an additional bug caused by the same underlying issue.